zio-http/src/main/scala/zhttp/http/Request.scala

```diff
@@ -103,15 +110,20 @@ object Request {
     remoteAddress: Option[InetAddress] = None,
     data: HttpData = HttpData.Empty,
   ): Request = {
-    val m  = method
-    val u  = url
-    val h  = headers
-    val ra = remoteAddress
-    val d  = data
+    val m        = method
+    val u        = url
+    val h        = headers
+    val ra       = remoteAddress
+    val d        = data
+    val jVersion = Version.`HTTP/1.1`.toJava
+    val path     = url.relative.encode
+    val jReq     = new DefaultFullHttpRequest(jVersion, method.toJava, path)
+
     new Request {
       override def method: Method                     = m
       override def url: URL                           = u
       override def headers: Headers                   = h
+      override def unsafeEncode: FullHttpRequest      = jReq
```

Comment:
create `DefaultFullHttpRequest` inside the `def`. So that it's only created when `unsafeEncode` is called.